phac-nml/irida-next
# Data Export - Create Export UI

## What does this PR do and why?
This PR adds the UI to create an export through the IRIDA Next website.

Note: The create modal currently only has the selected sample count. A sample listing will be added in a future PR.

## Screenshots or screen recordings
[Screencast from 2024-04-05 12:13:01 PM.webm](https://github.com/phac-nml/irida-next/assets/82407232/3ba855a5-e9d3-4841-9d49-9b19ff9ff1cd)

## How to set up and validate locally
1. Login with a user that has access level of analyst or higher
2. Navigate to the Groups samples page
3. Ensure no samples are selected and that the `Create Export` button is in a disabled state
4. Select any number of samples. The `Create Export` button should be enabled. Click the `Create Export` button
5. In the modal, ensure the sample count is accurate. Give the export a name, and then submit. You should be redirected to the Data Exports index view.
6. Repeat the above from the Project samples page

Additional tests:
1. Ensure selecting samples from separate group/project samples pages do not affect one another when creating an export
2. Ensure the data export preview content and download content are as expected
3. Ensure a user with access level of guest cannot see the `Create Export` button

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
